src/wallet/bdb.cpp

```diff
@@ -738,27 +739,30 @@ void BerkeleyDatabase::ReloadDbEnv()
     }
 }
 
-Dbc* BerkeleyBatch::GetCursor()
+bool BerkeleyBatch::StartCursor()
 {
+    assert(!m_cursor);
     if (!pdb)
-        return nullptr;
-    Dbc* pcursor = nullptr;
-    int ret = pdb->cursor(nullptr, &pcursor, 0);
-    if (ret != 0)
-        return nullptr;
-    return pcursor;
+        return false;
+    int ret = pdb->cursor(nullptr, &m_cursor, 0);
+    return ret == 0;
 }
 
-int BerkeleyBatch::ReadAtCursor(Dbc* pcursor, CDataStream& ssKey, CDataStream& ssValue)
+bool BerkeleyBatch::ReadAtCursor(CDataStream& ssKey, CDataStream& ssValue, bool& complete)
 {
+    complete = false;
+    if (m_cursor == nullptr) return false;
```

Comment:
`assert(m_cursor)`? There was no check for `pcursor`.

Reply:
I don't think an assert is necessary.

Reply:
If it's something that can't happen then an assertion is fine. Also, it means that `StartCursor` must be called before.

Reply:
If I have to push again.